File: Python3/ITP1_05_B.py
while True:
    n = [int(i) for i in input().split()]
    if n[0] == 0 and n[1] == 0:
        break
    for j in range(0, n[1]):
        print("#", end="")
    print()
    for i in range(1, n[0]-1):
        print("#", end=""),
        for j in range(1, n[1]-1):
            print(".", end="")
        print("#")
    for j in range(0, n[1]):
        print("#", end="")
    print()
    print()

File: Ruby/ITP1_05_B.rb
while true
  a = gets.split(" ").map(&:to_i)
  if a[0] == 0 && a[1] == 0 then
    break
  end
  a[1].times{|j|
    print "#"
  }
  puts ""
  (a[0]-2).times{|i|
    print "#"
    (a[1]-2).times{|j|
      print "."
    }
    puts "#"
  }
  a[1].times{|j|
    print "#"
  }
  puts ""
  puts ""
end
